pkg/plugins/utils/ci/main_test.go

```diff
@@ -0,0 +1,63 @@
+package ci
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNew(t *testing.T) {
+
+	tests := []struct {
+		name    string
+		env     map[string]string
+		expect  CIEngine
+		wantErr error
+	}{
+		{
+			name:    "Unknown",
+			wantErr: fmt.Errorf("Unknown CI Engine."),
```

Comment:
This test can't work from GitHub Action because `GITHUB_ACTION` is already set so we'll never reach  the final return

Reply:
And the Jenkins and Gitlab test will only work as long as we don't reorder the if conditions

Reply:
ooooh good catch! I need to unset variables prior to the tests \o/ Thanks!

Reply:
I wouldn't "unset" environment varialbe used by CI as it may be hard in the futur to debug CI issues
 I think in this case it's just fine to no test it